marxs/optics/polarization.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+
+from ..math.pluecker import *
+
+
+def polarization_vectors(dir_array, angles):
+	n = len(angles)
+	polarization = np.zeros((n, 4))
+	x = np.array([1., 0., 0.])
+	y = np.array([0., 1., 0.])
+	for i in range(0, n):
+		r = h2e(dir_array[i])
+		r /= np.linalg.norm(r)
+		if not (np.isclose(r[0], 0.) and np.isclose(r[2], 0.)):
+			# polarization relative to positive y at 0              
```

Comment:
I see why you have chosen to define it this way and in practice it might work out. In theory, I'm a little cautious: What happens if we change the direction of the photon e.g. in a small angle reflection or some scatter and suddenly it jumps from one definition to the other? At the very least we need to explain this in the documentation string of this function.

I don't have a better idea to define the polarization, I admit...